tracer/src/Datadog.Trace.Tools.Runner/RunCiCommand.cs

```diff
@@ -3,23 +3,285 @@
+            }
+
+            // If the agentless feature flag is enabled, we check for ApiKey
+            // If the agentless feature flag is disabled, we check if we have connection to the agent before running the process.
+            AgentConfiguration agentConfiguration = null;
+            IDiscoveryService discoveryService = NullDiscoveryService.Instance;
+            if (agentless)
+            {
+                Log.Debug("RunCiCommand: Agentless has been enabled. Checking API key");
+                if (string.IsNullOrWhiteSpace(apiKey))
+                {
+                    Utils.WriteError("An API key is required in Agentless mode.");
+                    Log.Error("RunHelper: An API key is required in Agentless mode.");
+                    return 1;
+                }
+            }
+            else
+            {
+                Log.Debug("RunCiCommand: Agent-based mode has been enabled. Checking agent connection.");
+                (agentConfiguration, discoveryService) = await Utils.CheckAgentConnectionAsync(settings.AgentUrl).ConfigureAwait(false);
+                if (agentConfiguration is null)
+                {
+                    Log.Error("RunCiCommand: Agent configuration cannot be retrieved.");
+                    return 1;
+                }
+            }
+
+            // Set Agentless configuration from the command line options
+            ciVisibilitySettings.SetAgentlessConfiguration(agentless, apiKey, applicationKey, ciVisibilitySettings.AgentlessUrl);
+
+            // Initialize flags to enable code coverage and test skipping
+            var codeCoverageEnabled = ciVisibilitySettings.CodeCoverageEnabled == true || ciVisibilitySettings.TestsSkippingEnabled == true;
+            var testSkippingEnabled = ciVisibilitySettings.TestsSkippingEnabled == true;
+
+            var createTestSession = false;
+            string codeCoveragePath = null;
+            var hasEvpProxy = !string.IsNullOrEmpty(agentConfiguration?.EventPlatformProxyEndpoint);
+            if (agentless || hasEvpProxy)
+            {
+                // Initialize CI Visibility with the current settings
+                Log.Debug("RunCiCommand: Initialize CI Visibility for the runner.");
+                CIVisibility.InitializeFromRunner(ciVisibilitySettings, discoveryService, hasEvpProxy);
+
+                // Upload git metadata
+                Log.Debug("RunCiCommand: Uploading repository changes.");
+                var itrClient = new IntelligentTestRunnerClient(CIEnvironmentValues.Instance.WorkspacePath, ciVisibilitySettings);
+                await itrClient.UploadRepositoryChangesAsync().ConfigureAwait(false);
+
+                // Once the repository has been uploaded we switch off the git upload in children processes
+                profilerEnvironmentVariables[Configuration.ConfigurationKeys.CIVisibility.GitUploadEnabled] = "0";
```

Comment:
It doesn't look like you checked this value before doing the upload - is that checked in `UploadRepositoryChangesAsync()`?

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3736/commits/4cb4314d519940f8ded24970a982c6fe44cad5b1